docs/db_migration_guide.md

```diff
@@ -0,0 +1,244 @@
+
+This guide explains the migration path, backend trade-offs, subsystem readiness, and best practices for safely upgrading your LND database.
+
+---
+
+## Operation Modes
+
+The LND database upgrade operates in two sequential stages, each with distinct tooling and implications:
+
+```mermaid
+flowchart TD
+    %% 1. Define all nodes
+    Bbolt["Bbolt (kvdb)"]
+    SQLite["SQLite (kvdb)"]
+    Postgres["Postgres (kvdb)"]
+    
+    M1_Sqlite["Migration #1 (lnd v0.19)<br>Invoices"]
+    M2_Sqlite["Migration #2 (lnd v0.20)<br>Graph"]
+    M3_Sqlite["Migration #3 (lnd v0.21)<br>Payments"]
+
+    M1_Postgres["Migration #1 (lnd v0.19)<br>Invoices"]
+    M2_Postgres["Migration #2 (lnd v0.20)<br>Graph"]
+    M3_Postgres["Migration #3 (lnd v0.21)<br>Payments"]
+
+    %% 2. Define all links (within and between graphs)
+    Bbolt --> SQLite
+    Bbolt --> Postgres
+    
+    SQLite --> M1_Sqlite
+    M1_Sqlite --> M2_Sqlite
+    M2_Sqlite --> M3_Sqlite
+
+    Postgres --> M1_Postgres
+    M1_Postgres --> M2_Postgres
+    M2_Postgres --> M3_Postgres
+
+    %% 3. Group nodes into subgraphs
+    subgraph "Step 1: Migration via lndinit"
+        Bbolt
+        SQLite
+        Postgres
+    end
+
+    subgraph "Step 2: Migration within lnd"
+        M1_Sqlite
+        M2_Sqlite
+        M3_Sqlite
+        M1_Postgres
+        M2_Postgres
+        M3_Postgres
+    end
+
+    %% 4. Apply Styles
+    style Bbolt fill:#e2e3e5,stroke:#383d41
+    style SQLite fill:#d4edda,stroke:#155724
+    style M1_Sqlite fill:#d4edda,stroke:#155724
+    style M2_Sqlite fill:#d4edda,stroke:#155724
+    style M3_Sqlite fill:#d4edda,stroke:#155724
+    style Postgres fill:#cce5ff,stroke:#004085
+    style M1_Postgres fill:#cce5ff,stroke:#004085
+    style M2_Postgres fill:#cce5ff,stroke:#004085
+    style M3_Postgres fill:#cce5ff,stroke:#004085
+
+```
+
+- **Stage 1**: Migrate from bbolt to a SQL-based **kvdb** backend using the [lndinit](https://github.com/lightninglabs/lndinit/blob/main/docs/data-migration.md) tool.
+- **Stage 2**: Incrementally migrate subsystem data (invoices, graph, payments, etc.) from **kvdb** to SQL native **relational tables** as support becomes available.
+
+---
+
+## Stage 1: Migration from bbolt to SQLite/Postgres (kvdb)
+
+LND cannot run with mixed backends, so all users must first leave bbolt behind. This stage uses the `lndinit` utility to perform an offline migration.
+
+### Choosing Your Target Backend
+
+| Backend   | Performance (in kvdb mode) | Default in LND? | Long-Term Viability |
+|-----------|--------------------------|------------------|----------------------|
+| Postgres  | Mediocre                 | No               | ✅ |
+| SQLite    | Good                     | **Yes (future)** | ✅ |
+
+> 💡 **Recommendation**: Unless you require Postgres for infrastructure reasons, **migrate to SQLite kvdb** as your interim backend.
```

Comment:
although, there is no current planned migration between the two.. so it isnt really an interim rn